File: js/leetcode/144-binary-tree-preorder-traversal.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
var preorderTraversal = function (root) {
  let output = []
  if (root) {
    _traverse(root, output);
  }
  return output
};

var _traverse = function (root, arr) {
  if (root) {
    arr.push(root.val)
    _traverse(root.left, arr);
    _traverse(root.right, arr);
  }
}

console.log(preorderTraversal([1, null, 2, 3])) // [1,2,3]

File: c#/leetcode/144-binary-tree-preorder-traversal.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution
{
  public IList<int> PreorderTraversal(TreeNode root)
  {
    var res = new List<int>();

    traverse(root, res);
    return res.ToArray();
  }

  public void traverse(TreeNode root, IList<int> res)
  {
    if (root != null)
    {
      res.Add(root.val);
      traverse(root.left, res);
      traverse(root.right, res);
    }
  }
}